examples/protocols/http_request/README.md

```diff
@@ -38,77 +42,207 @@ See the Getting Started Guide for full steps to configure and use ESP-IDF to bui
+I (9474) example: 6... 
+I (10474) example: 5... 
+I (11474) example: 4... 
+I (12474) example: 3... 
+I (13474) example: 2... 
+I (14474) example: 1... 
+I (15474) example: 0... 
+I (16474) example: Starting again!
+I (16474) example: DNS lookup v4v6.ipv6-test.com (port 80)
+I (16474) example_common: IPv6 DNS lookup
+I (16514) example: DNS lookup succeeded. (10) IP=2001:41D0:701:1100::29C8
+I (16514) example: ... allocated socket
+I (16924) example: ... connected
+I (16924) example: ... socket send success
+I (16924) example: ... set socket receiving timeout success
+HTTP/1.1 200 OK
+Date: Sat, 24 Feb 2024 05:33:19 GMT
+Server: Apache/2.4.25 (Debian)
+Vary: Accept-Encoding
+Connection: close
+Content-Type: text/html; charset=UTF-8
+
+2407:8800:bc61:1340:a3a:f2ff:fe65:db28
+I (17334) example: ... done reading from socket. Last read return=0 errno=128.
+I (17344) example: 10... 
+```
+
+On a dual-stack network when connecting to an IPv4-only server, then (with an IPv6 global address) an IPv6 lookup is still attempted first. If DNS64/NAT64 is not available, then the IPv6 DNS lookup will fail, and the code will fall back to an IPv4 DNS lookup.
+
+```
+I (17034) example: DNS lookup v4.ipv6-test.com (port 80)
+I (17034) example_common: IPv6 DNS lookup
+I (17074) example_common: - IPv6 DNS lookup failed, trying IPv4 lookup
+I (17074) example_common: IPv4 DNS lookup
+I (17074) example: DNS lookup succeeded. (2) IP=51.75.78.103
+I (17084) example: ... allocated socket
+I (17694) example: ... connected
+I (17694) example: ... socket send success
+I (17694) example: ... set socket receiving timeout success
+HTTP/1.1 200 OK
+Date: Sat, 24 Feb 2024 06:10:00 GMT
+Server: Apache/2.4.25 (Debian)
+Vary: Accept-Encoding
+Connection: close
+Content-Type: text/html; charset=UTF-8
+
+220.240.255.134
+I (18314) example: ... done reading from socket. Last read return=0 errno=128.
+I (18314) example: 10...
 ```
-I (10557) example_connect: - IPv4 address: 192.168.194.219
-I (10557) example_connect: - IPv6 address: fe80:0000:0000:0000:266f:28ff:fe80:2c74, type: ESP_IP6_ADDR_IS_LINK_LOCAL
-W (10577) wifi:<ba-add>idx:0 (ifx:0, ee:6d:19:60:f6:0e), tid:0, ssn:3, winSize:64
-I (10587) example: DNS lookup succeeded. IP=93.184.216.34
-I (10587) example: ... allocated socket
-I (10917) example: ... connected
-I (10917) example: ... socket send success
-I (10927) example: ... set socket receiving timeout success
-HTTP/1.0 200 OK
-Age: 317271
-Cache-Control: max-age=604800
+
+### IPv4-only network connecting to dual-stack server
+
+This shows an IPv4 only network (Shadow) connecting to a dual-stack server (`v4v6.ipv6-test.com`).
+
+The connection attempt starts as soon as the IPv4 address is received, and as the device has no global IPv6 address, the DNS lookup and connection will only be IPv4 (`51.75.78.103`).
+
+Note that although the device has an IPv6 link-local address it is ignored, as it is not usable to connect to a public address returned from DNS.
```

Comment:
```suggestion
Note that, although the device has an IPv6 link-local address, it is ignored because it cannot be used to connect to a public address returned from DNS
```

Reply:
Done. Fix included and rebased to lastest head.